nixos/modules/config/networking.nix

```diff
@@ -9,7 +9,7 @@ let
   cfg = config.networking;
   dnsmasqResolve = config.services.dnsmasq.enable &&
                    config.services.dnsmasq.resolveLocalQueries;
-  hasLocalResolver = config.services.bind.enable || dnsmasqResolve;
+  hasLocalResolver = cfg.nameservers == [] && ( config.services.bind.enable || dnsmasqResolve );
```

Comment:
What exactly happens here when `nameservers` is *not* empty?

Reply:
The intention of this PR (apparently): only use nameservers from the explicit `nameservers` list and not `127.0.0.1`.

Side note: I added another resolver months ago and didn't realize we had this magic; I should amend `hasLocalResolver` for that.

Reply:
@peti: When 'nameservers' is empty, it follows the old behaviour. That is: it sets 'nameserver = 127.0.0.1' in networking.nix when either services.bind.enable == true or services.dnsmasq.resolveLocalQueries == true at https://github.com/gwitmond/nixpkgs/blob/4b8c7065e2d9f2961855e7ed2cd51b4bbb6d5f11/nixos/modules/config/networking.nix#L241

If neither networking.nameservers is set nor either of those services, it defaults to standard behaviour.

@vcunat: Indeed, my goal is to give the user specified nameservers priority over any automatic ones.